hail/python/hailtop/hailctl/dataproc/start.py

```diff
@@ -326,6 +335,45 @@ def disk_size(size):
     conf.flags['secondary-worker-boot-disk-size'] = disk_size(args.secondary_worker_boot_disk_size)
     conf.flags['worker-boot-disk-size'] = disk_size(args.worker_boot_disk_size)
     conf.flags['worker-machine-type'] = args.worker_machine_type
+
+    if not args.no_off_heap_memory:
+        worker_memory = MACHINE_MEM[args.worker_machine_type]
+
+        # A Google support engineer recommended the strategy of passing the YARN
+        # config params, and the default value of 95% of machine memory to give to YARN.
+        # yarn.nodemanager.resource.memory-mb - total memory per machine
+        # yarn.scheduler.maximum-allocation-mb - max memory to allocate to each container
+        available_memory_fraction = args.yarn_memory_fraction
+        available_memory_mb = int(worker_memory * available_memory_fraction * 1024)
+        cores_per_machine = int(args.worker_machine_type.split('-')[-1])
+        executor_cores = min(cores_per_machine, 4)
```

Comment:
Confused about the min here. Is it the case that when using larger worker machines, we assign more executors per machine and always end up with 4 cores per executor?

Reply:
that's correct, yes. Larger executors than 4 cores tend to slow things down because there's contention for shared resources (broadcasts, static object stuff, etc).

This keeps the same semantics that we've had, but makes it explicit.